lms/djangoapps/courseware/user_state_client.py

```diff
@@ -193,6 +193,11 @@ def set_many(self, username, block_keys_to_state, scope=Scope.user_state):
         else:
             user = User.objects.get(username=username)
 
+        if user.is_anonymous():
```

Comment:
To be consistent with the other places where we avoid user-checking, should we check here for `getattr(user, 'known')` instead?  @ormsbee